### PR TITLE
docs(examples): Add NestJS example

### PR DESCRIPTION
**Pull request by bendechrai**
Uses Node.js SDK
Uses a Nest Guard with external default Arcjet configuration file Allows each controller to augment the Arcjet configuration

### CODE REVIEW

File: examples/nodejs-nestjs/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  <picture>
+    <source media="(prefers-color-scheme: dark)" srcset="https://arcjet.com/arcjet-logo-minimal-dark-mark-all.svg">
+    <img src="https://arcjet.com/arcjet-logo-minimal-light-mark-all.svg" alt="Arcjet Logo" height="128" width="auto">
+  </picture>
+</a>
+
+# Arcjet Protection with NestJS for Node.js
+
+This example shows how to use Arcjet to protect [NestJS](https://nestjs.com/) apps using the Node.js SDK.
+
```
Comment (davidmytton) [inline examples/nodejs-nestjs/README.md:11]: As this is a bit different from how our other examples work with the rules all defined in a single place on the instantiated client, I think it's worth adding some more explanation. Perhaps point out where our SDK is integrated and how to add more rules.

File: examples/nodejs-nestjs/src/app.module.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import { AppService } from './app.service.js';
+
+import { ProtectedModule } from './protected/protected.module.js';
+
+import { ArcjetGuard } from './arcjet/arcjet.guard.js';
+
+@Module({
+  imports: [ConfigModule.forRoot(), ProtectedModule],
+  controllers: [AppController],
+  providers: [
```
Comment (davidmytton) [inline examples/nodejs-nestjs/src/app.module.ts:14]: Add a comment here about what this does, why Arcjet is integrated here, and a link to the NestJS docs related to this?

File: examples/nodejs-nestjs/src/arcjet/arcjet.guard.ts [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+@Injectable()
+export class ArcjetGuard implements CanActivate {
+    private aj: any;
+
+    constructor(
+        private configService: ConfigService,
+        @Optional() @Inject('ARCJET_RULES') private rules: any[] = []
+    ) {
+        this.aj = arcjet({
```
Comment (davidmytton) [inline examples/nodejs-nestjs/src/arcjet/arcjet.guard.ts:14]: Worth a brief comment explaining what this is doing and where to set the rules.

File: examples/nodejs-nestjs/src/arcjet/arcjet.guard.ts [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+            rules: [...ARCJET_DEFAULT_RULES, ...this.rules],
+        });
+    }
+
+    async canActivate(
+        context: ExecutionContext,
+    ): Promise<boolean> {
+        const request = context.switchToHttp().getRequest();
+
+        // const aj = arcjet({
```
Comment (davidmytton) [inline examples/nodejs-nestjs/src/arcjet/arcjet.guard.ts:25]: I suggest explaining why this is commented out and when/why you might want to use it.

File: examples/nodejs-nestjs/src/config/arcjet.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { detectBot } from '@arcjet/node';
+
+export const ARCJET_DEFAULT_RULES = [
```
Comment (davidmytton) [inline examples/nodejs-nestjs/src/config/arcjet.ts:3]: Also worth a comment explaining this and how you might adapt it to your own use.

File: examples/nodejs-nestjs/src/protected/protected.module.ts [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+import { fixedWindow } from '@arcjet/node';
+import { ArcjetGuard } from '../arcjet/arcjet.guard.js';
+import { ConfigModule } from '@nestjs/config';
+import { ProtectedService } from './protected.service.js';
+
+@Module({
+    imports: [ConfigModule.forRoot()],
+    controllers: [ProtectedController],
+    providers: [
+        ProtectedService,
```
Comment (davidmytton) [inline examples/nodejs-nestjs/src/protected/protected.module.ts:12]: Add a comment here explaining that you can add different / chained rules and link to our docs?

File: examples/nodejs-nestjs/package.json [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+    "start:prod": "node dist/main",
+    "lint": "eslint \"{src,apps,libs,test}/**/*.ts\" --fix",
+    "test": "jest",
+    "test:watch": "jest --watch",
+    "test:cov": "jest --coverage",
+    "test:debug": "node --inspect-brk -r tsconfig-paths/register -r ts-node/register node_modules/.bin/jest --runInBand",
+    "test:e2e": "jest --config ./test/jest-e2e.json"
+  },
+  "dependencies": {
+    "@arcjet/node": "^1.0.0-alpha.12",
```
Comment (davidmytton) [inline examples/nodejs-nestjs/package.json:25]: I think we'll want to add this to CI so this should be a file reference so it always pulls the latest SDK from the repo. Example at https://github.com/arcjet/arcjet-js/blob/330b3178a2ac273134d5a18d60078d651bd72f67/examples/nextjs-14-decorate/package.json#L13

**Review by davidmytton — CHANGES_REQUESTED** [review]
Great to see this integrated with the native NestJS functionality! All my comments are about adding comments 😅 to explain what it's doing and how the developer could adapt to their own use case.

File: examples/nodejs-nestjs/src/config/arcjet.ts [hunk 8]
```diff
@@ -0,0 +1,7 @@
...
+    detectBot({
+        mode: "LIVE",
+    }),
```
Comment (blaine-arcjet) [inline examples/nodejs-nestjs/src/config/arcjet.ts:6]: I think we should stick with `shield` being the default rule and all others people add.

File: examples/nodejs-nestjs/src/app.service.ts [hunk 9]
```diff
@@ -0,0 +1,8 @@
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export class AppService {
+  getHello(): string {
+    return 'This endpoint is protected by bot detection. Go to /protected to test the addition of rate limiting.';
```
Comment (blaine-arcjet) [inline examples/nodejs-nestjs/src/app.service.ts:6]: Is this actually protected? I don't see the use of `@UseGuards(ArcjetGuard)`

File: examples/nodejs-nestjs/src/config/arcjet.ts (hunk 8 above)
Comment (davidmytton) [reply examples/nodejs-nestjs/src/config/arcjet.ts:10]: Agreed on this. Let's swap it for `shield` to protect every route and get them to test the rate limit endpoint as part of the flow with a low limit.

**Review by davidmytton — APPROVED** [review]

**Comment by blaine-arcjet** [comment]
@SocketSecurity ignore-all

**Review by blaine-arcjet — APPROVED** [review]

**Pull request merged**